### PR TITLE
Cj/action item form

### PR DESCRIPTION
**Pull request by christopher-grey**
## Action Item Modal/Form/Cards

Describe your feature here
Created action item modal with card list and action item form button
### Related PRs
Action Item Form
Optional - any related PRs you're waiting on, or PRs that will conflict, etc

### Migrations

Optional - if you added anything to the database through migration(s)

### Screenshots

Optional but would be nice for final presentation :)

CC: @didvi

### CODE REVIEW

File: app/javascript/components/ActionItemCard/index.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import NoteSharpIcon from '@material-ui/icons/NoteSharp';
+import HouseIcon from '@material-ui/icons/House';
+import EcoSharpIcon from '@material-ui/icons/EcoSharp';
+import CreateSharpIcon from '@material-ui/icons/CreateSharp';
+import SentimentSatisfiedSharpIcon from '@material-ui/icons/SentimentSatisfiedSharp';
+import CodeRoundedIcon from '@material-ui/icons/CodeRounded';
+import styles from './styles';
+class ActionItemCard extends React.Component {
+  getIconComponent(Icon) {
+    console.log(Icon);
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemCard/index.js:19]: rm print statement

File: app/javascript/components/ActionItemForm/index.js [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+  Button,
+  TextField,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Grid,
+} from '@material-ui/core';
+import { withStyles, MuiThemeProvider } from '@material-ui/core/styles';
+
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:22]: rm line

File: app/javascript/components/ActionItemForm/index.js [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+      type: this.props.type,
+      title: this.props.title,
+      open: false,
+      dueDate: this.props.dueDate,
+      category: this.props.category,
+      errors: {
+        title: '',
+      },
+    };
+    // this.onChange = editorState => this.setState({ editorState });
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:40]: rm comment

File: app/javascript/components/ActionItemForm/index.js [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    this.setState({ open: true });
+  }
+
+  handleClose() {
+    this.setState({
+      open: false,
+      title: this.props.title,
+    });
+    if (this.state.type === 'edit') {
+      (this.state.title = this.props.title)(
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:59]: we can rm the parentheses in these two lines

File: app/javascript/components/CaseNoteCard/index.js [hunk 5]
```diff
@@ -118,6 +118,9 @@ class CaseNoteCard extends React.Component {
                     description={this.state.description}
                     title={this.state.title}
                   />
+                  {/* <ActionItemModal
```
Comment (wilsonw926) [inline app/javascript/components/CaseNoteCard/index.js:121]: rm this

File: app/javascript/components/ActionItemForm/index.js [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+
+  checkErrors = field => () => {
+    const errorMessage = '';
+    if (field === 'title') {
+      const { title } = this.state;
+      if (
+        title === '' ||
+        validator.isEmpty(title, { ignore_whitespace: true })
+      ) {
+        // eslint-disable-next-line
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:73]: rm this

File: app/javascript/components/ActionItemForm/index.js [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+      title: this.props.title,
+    });
+    if (this.state.type === 'edit') {
+      (this.state.title = this.props.title)(
+        (this.state.description = this.props.description),
+      );
+    }
+  }
+
+  checkErrors = field => () => {
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:65]: For this file, either keep functions as arrow notation like this function checkErrors or use the format as handleClose(). either are fine but keep it consistent.

File: app/javascript/components/ActionItemForm/index.js [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+    this.setState({ [name]: value });
+  };
+
+  handleDescriptionChange = name => state => {
+    const value = JSON.stringify(convertToRaw(state.getCurrentContent()));
+    this.setState({ [name]: value });
+  };
+
+  handleCategoryChange = name => () => {
+    console.log(name);
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:93]: rm print statement

File: app/javascript/components/ActionItemForm/index.js [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+            <TextField
+              value={this.state.dueDate || ''}
+              style={styles.dialogContentTextFieldStyle}
+              name="Due Date"
+              onChange={this.handleChange('dueDate')}
+              onBlur={this.checkErrors('Due Date')}
+              variant="outlined"
+              margin="dense"
+              id="Due Date"
+              // label="date when assignment is due "
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemForm/index.js:285]: rm

File: app/javascript/components/Assignments/index.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+// import React from 'react';
```
Comment (wilsonw926) [inline app/javascript/components/Assignments/index.js:1]: can we rm this file?

File: app/javascript/components/ActionItemModal/index.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+      dueDate,
+      category,
+    };
+    this.setState(prevState => ({
+      cardList: [...prevState.cardList, newCard],
+    }));
+  }
+
+  render() {
+    const cards = this.state.cardList.map(card => (
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemModal/index.js:33]: niceee

File: app/javascript/components/ActionItemModal/index.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+import { withStyles } from '@material-ui/core/styles';
+import ActionItemCard from 'components/ActionItemCard';
+import ActionItemForm from 'components/ActionItemForm';
+import styles from './styles';
+class ActionItemModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cardList: [],
+      //  cardList: {
```
Comment (wilsonw926) [inline app/javascript/components/ActionItemModal/index.js:11]: rm this commented out code

File: app/javascript/components/ActionItemCard/index.js [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+          item
+          xs
+          container
+          direction="row"
+          justify="flex-start"
+          alignItems="flex-start"
+        >
+          <Grid item xs>
+            <Typography gutterBottom variant="subtitle1">
+              title = {this.props.title}
```
Comment (didvi) [inline app/javascript/components/ActionItemCard/index.js:54]: Just display the title, dont put "title =" in front of each title

File: app/javascript/components/ActionItemCard/index.js [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+          xs
+          container
+          direction="row"
+          justify="space-evenly"
+          alignItems="flex-start"
+          spacing={2}
+        >
+          <Grid item xs>
+            <Typography variant="body2" style={{ cursor: 'pointer' }}>
+              dueDate = {this.props.dueDate}
```
Comment (didvi) [inline app/javascript/components/ActionItemCard/index.js:81]: remove the "dueDate ="

File: app/javascript/components/ActionItemForm/index.js [hunk 15]
```diff
@@ -0,0 +1,373 @@
...
+      </Button>
+    );
+    return ret;
+  };
+
+  render() {
+    const { classes } = this.props;
+
+    const categories = ["Finances", "Project", "Community", "Startup", "Treatment", "Health", "Education"]
+    console.log(classes)
```
Comment (didvi) [inline app/javascript/components/ActionItemForm/index.js:137]: remove

**Pull request closed without merging**